### PR TITLE
[WIP]add module pos_membership.

### PR DESCRIPTION
**Pull request by Rona111**
cc @elicoidal @duanyp1991 @siyuancn @seb-elico

### CODE REVIEW

File: pos_membership/LICENSE [hunk 1]
```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```
Comment (elicoidal) [inline pos_membership/LICENSE]: no need this file

File: pos_membership/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+============
+
+ You need to have the following modules available:
+    * membership
+    * pos_pricelist (optional) -> to allow VIP members have a different price.
+    * membership_account_balance
+
+Usage
+=====
+ 1- create a new payment method / account journal: VIP card
```
Comment (elicoidal) [inline pos_membership/README.rst]: use standard enumerated lists: 
#.

File: pos_membership/models/models.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            if self._context.get('invoice_id'):
+                invoice = self.env['account.invoice'].browse(
+                    self._context['invoice_id'])
+                if invoice._is_there_membership_product():
+                    raise Warning(
+                        _('Cannot use VIP card for membership product!'))
+
+            # check if there is enough membership balance
+            if not self._check_enough_membership_balance():
+                raise Warning(_('No enough membership balance!'))
```
Comment (elicoidal) [inline pos_membership/models/models.py]: not enough

File: pos_membership/static/src/js/screens.js [hunk 4]
```diff
@@ -0,0 +1,340 @@
+function openerp_pos_membership (instance, module){
```
Comment (elicoidal) [inline pos_membership/static/src/js/screens.js:5]: copyright

File: pos_membership/tests/test_membership_account_balance_future.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+import openerp.tests.common as common
+
+
+class TestMembershipAccountBalanceFuture(common.TransactionCase):
+    def setUp(self):
+        super(TestMembershipAccountBalanceFuture, self).setUp()
+        # partner to be used to test.
+        self.partner = self.env['res.partner'].create({
+            'name': 'Alex Duan',
```
Comment (elicoidal) [inline pos_membership/tests/test_membership_account_balance_future.py]: generic name

**Comment by elicoidal** [comment]
LGTM (small details).
Could be propose to the OCA

**Comment by siyuancn** [comment]
LGTM

**Comment by elicoidal** [comment]
@Rona111 cleanse the small details and we merge

**Comment by Rona111** [comment]
cc @elicoidal @siyuancn @duanyp1991

**Comment by elicoidal** [comment]
:+1:

**Pull request merged**